qt/python/mantidqtinterfaces/mantidqtinterfaces/PyChop/PyChopGui.py

```diff
@@ -214,22 +214,20 @@ def setFreq(self, freqtext=None, **kwargs):
         # Checks for independent phases
         phases = []
         for key, widget in self.widgets.items():
-            if "MERLIN" in str(self.engine.instname) and key.endswith("Phase") and widget["Label"].isHidden():
+            if key.endswith("Phase"):
                 # Special case for MERLIN
                 # sets the default phase for Chopper0Phase if not in "Instrument Scientist Mode"
-                phase = float(widget["Edit"].text())
-                phases.append(phase)
-            if key.endswith("Phase") and not widget["Label"].isHidden():
-                idx = int(key[7])
-                phase = widget["Edit"].text()
-                if isinstance(self.engine.chopper_system.defaultPhase[idx], str):
-                    phase = str(phase)
-                else:
-                    try:
-                        phase = float(phase) % (1e6 / self.engine.moderator.source_rep)
-                    except ValueError:
-                        raise ValueError(f'Incorrect phase value "{phase}" for {widget["Label"].text()}')
-                phases.append(phase)
+                if not widget["Label"].isHidden() or "MERLIN" in str(self.engine.instname) and widget["Label"].isHidden():
```

Comment:
You can simplify the code to:

```python
if not widget["Label"].isHidden() or "MERLIN" in str(self.engine.instname)
```

Because the logic is `not A or (B and A)` which is the same as `(not A or B) and (not A or A)` and the second part of that will always evaluate to `True`.

Reply:
That is correct. I just thought it would make it clearer in what respect MERLIN is different from the normal case.
I will update it.

Also, I am currently getting a matplotlib error when changing the value for Ei for the third time. It might have something to do with my local environment though, but I want to be sure before pushing another commit.

Reply:
The matplotlib error was indeed caused by my local setup and not the code changes for this PR. However, I had to limit the special case for MERLIN to Chopper0Phase, as otherwise the unit tests would fail.